# Introduction

With this data set, we will examine the gold export structure of Ghana and Nigeria worldwide. The aim is to evaluate the gold export trends, volumes and impact of these two countries on the world market. Dataset gathered form UN Comtrade.

## Dataset Columns & Definitions

**TypeCode**: Identifies the type of record, typically indicating whether it's commercial goods, services, etc.

**FreqCode**: Indicates the frequency of the data report (e.g., 'M' for monthly, 'A' for annual).

**RefPeriodId**: A unique identifier for the reference period of the data, often combining year and month.

**RefYear**: The year to which the data pertains.

**RefMonth**: The month to which the data pertains.

**Period**: A combined year and month format that specifies the exact period of the data report.

**ReporterCode**: Numeric code representing the country that reported the data.

**ReporterISO**: The ISO code of the reporting country.

**ReporterDesc**: Descriptive name of the reporting country.

**FlowCode**: Code indicating the direction of trade flow (e.g., exports or imports).

**FlowDesc**: Description of the trade flow type.

**PartnerCode**: Numeric code representing the partner country in the trade.

**PartnerISO**: The ISO code of the partner country.

**PartnerDesc**: Descriptive name of the partner country.

**Partner2Code**: Numeric code for a secondary partner country, if applicable.

**Partner2ISO**: ISO code of a secondary partner country.

**Partner2Desc**: Description of a secondary partner country.

**ClassificationCode**: Code of the classification used for the products or services in the report.

**ClassificationSearchCode**: A searchable version of the classification code.

**IsOriginalClassification**: Boolean indicating whether the classification is the original one used by the reporting country.

**CmdCode**: Code for the commodity or service reported.

**CmdDesc**: Description of the commodity or service.

**AggrLevel**: Level of aggregation of the data.

**IsLeaf**: Boolean indicating if the data point is at the lowest level of the classification tree.

**CustomsCode**: Code used by customs for the reported items.

**CustomsDesc**: Description of the customs code.

**MosCode**: Code indicating the mode of shipment.

**MotCode**: Mode of transport code.

**MotDesc**: Description of the mode of transport.

**QtyUnitCode**: Code for the unit of quantity in which the goods are measured.

**QtyUnitAbbr**: Abbreviation of the quantity unit.

**Qty**: Quantity of the goods traded.

**IsQtyEstimated**: Boolean indicating whether the quantity is estimated.

**AltQtyUnitCode**: Code for an alternate unit of quantity.

**AltQtyUnitAbbr**: Abbreviation for the alternate unit of quantity.

**AltQty**: Alternate quantity measurement, if used.

**IsAltQtyEstimated**: Boolean indicating whether the alternate quantity is estimated.

**NetWgt**: Net weight of the goods traded.

**IsNetWgtEstimated**: Boolean indicating whether the net weight is estimated.

**GrossWgt**: Gross weight of the goods traded.

**IsGrossWgtEstimated**: Boolean indicating whether the gross weight is estimated.

**Cifvalue**: CIF (Cost, Insurance, Freight) value of the goods imported.

**Fobvalue**: FOB (Free on Board) value of the goods exported.

**PrimaryValue**: The primary monetary value of the trade.

**LegacyEstimationFlag**: Indicates if the values are based on legacy estimation methods.

**IsReported**: Boolean indicating whether the data was directly reported by the country.

**IsAggregate**: Boolean indicating whether the data is an aggregation of multiple records.

## Libraries

In [1]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

### Data Import

In [2]:
# In local, I identify the current path of dataset
file_path = r"C:\Users\MRE\Desktop\Data Science & Business\Project\Dataset\aga_bu_son_olsun_artik.csv"

# Then I upload
df = pd.read_csv(file_path, encoding='iso-8859-1')

In [3]:
# First look to dataset
df.head()

TypeCode FreqCode  RefPeriodId  RefYear  RefMonth  Period  ReporterCode  \
0        C        A     20170101     2017        52    2017           288   
1        C        A     20170101     2017        52    2017           288   
2        C        A     20170101     2017        52    2017           288   
3        C        A     20170101     2017        52    2017           288   
4        C        A     20170101     2017        52    2017           288   

  ReporterISO ReporterDesc FlowCode  ... IsNetWgtEstimated  GrossWgt  \
0         GHA        Ghana        X  ...              True       NaN   
1         GHA        Ghana        X  ...             False       NaN   
2         GHA        Ghana        X  ...             False       NaN   
3         GHA        Ghana        X  ...             False       NaN   
4         GHA        Ghana        X  ...             False       NaN   

  IsGrossWgtEstimated Cifvalue      Fobvalue  PrimaryValue  \
0               False      NaN  5.858281e+09  5.858281e+09   
1               False      NaN  1.103010e+06  1.103010e+06   
2               False      NaN  9.274560e+05  9.274560e+05   
3               False      NaN  4.926127e+03  4.926127e+03   
4               False      NaN  1.940124e+04  1.940124e+04   

  LegacyEstimationFlag IsReported IsAggregate  Unnamed: 47  
0                    0      False        True          NaN  
1                    0      False        True          NaN  
2                    0      False        True          NaN  
3                    0      False        True          NaN  
4                    0      False        True          NaN  

[5 rows x 48 columns]

## Data Exploration

In [4]:
# Let's look the shape of dataset
df.shape

(368, 48)

* As we can see here, dataset has 368 rows, and 48 columns.

In [16]:
df.dtypes

TypeCode                     object
FreqCode                     object
RefPeriodId                   int64
RefYear                       int64
RefMonth                      int64
Period                        int64
ReporterCode                  int64
ReporterISO                  object
ReporterDesc                 object
FlowCode                     object
FlowDesc                     object
PartnerCode                   int64
PartnerISO                   object
PartnerDesc                  object
Partner2Code                  int64
Partner2ISO                  object
Partner2Desc                 object
ClassificationCode           object
ClassificationSearchCode     object
IsOriginalClassification       bool
CmdCode                       int64
CmdDesc                      object
AggrLevel                     int64
IsLeaf                         bool
CustomsCode                  object
CustomsDesc                  object
MosCode                       int64
MotCode                     

* We can see the data tpyes for all columns.

In [17]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 368 entries, 0 to 367
Data columns (total 48 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   TypeCode                  368 non-null    object 
 1   FreqCode                  368 non-null    object 
 2   RefPeriodId               368 non-null    int64  
 3   RefYear                   368 non-null    int64  
 4   RefMonth                  368 non-null    int64  
 5   Period                    368 non-null    int64  
 6   ReporterCode              368 non-null    int64  
 7   ReporterISO               368 non-null    object 
 8   ReporterDesc              368 non-null    object 
 9   FlowCode                  368 non-null    object 
 10  FlowDesc                  368 non-null    object 
 11  PartnerCode               368 non-null    int64  
 12  PartnerISO                368 non-null    object 
 13  PartnerDesc               368 non-null    object 
 14  Partner2Co

* Also we can clearly see with non-null count plus data types for all columns

In [42]:
df.describe()

RefPeriodId      RefYear  RefMonth       Period  ReporterCode  \
count  3.680000e+02   368.000000     368.0   368.000000    368.000000   
mean   2.019736e+07  2019.725543      52.0  2019.725543    318.217391   
std    2.013091e+04     2.013091       0.0     2.013091     86.647248   
min    2.017010e+07  2017.000000      52.0  2017.000000    288.000000   
25%    2.018010e+07  2018.000000      52.0  2018.000000    288.000000   
50%    2.019010e+07  2019.000000      52.0  2019.000000    288.000000   
75%    2.022010e+07  2022.000000      52.0  2022.000000    288.000000   
max    2.023010e+07  2023.000000      52.0  2023.000000    566.000000   

       PartnerCode  Partner2Code        CmdCode   AggrLevel  MosCode  ...  \
count   368.000000         368.0     368.000000  368.000000    368.0  ...   
mean    494.831522           0.0  401029.627717    5.119565      0.0  ...   
std     283.515473           0.0  349803.862087    0.994178      0.0  ...   
min       0.000000           0.0    7108.000000    4.000000      0.0  ...   
25%     276.000000           0.0    7108.000000    4.000000      0.0  ...   
50%     586.000000           0.0  710812.000000    6.000000      0.0  ...   
75%     757.000000           0.0  710813.000000    6.000000      0.0  ...   
max     854.000000           0.0  710813.000000    6.000000      0.0  ...   

                Qty  AltQtyUnitCode        AltQty         NetWgt  GrossWgt  \
count  3.680000e+02      368.000000  3.680000e+02     356.000000     239.0   
mean   1.102788e+05        7.364130  1.377325e+05    9119.724034       0.0   
std    4.705406e+05        2.309326  6.304052e+05   26926.547545       0.0   
min    0.000000e+00       -1.000000  0.000000e+00       0.000000       0.0   
25%    1.870000e+00        8.000000  1.840000e+00       1.060000       0.0   
50%    5.375000e+01        8.000000  5.187000e+01      13.470000       0.0   
75%    3.217799e+03        8.000000  2.329050e+03     688.179000       0.0   
max    3.816019e+06        8.000000  5.822523e+06  242131.663000       0.0   

       Cifvalue      Fobvalue  PrimaryValue  LegacyEstimationFlag  Unnamed: 47  
count      17.0  3.680000e+02  3.680000e+02            368.000000          0.0  
mean        0.0  4.105521e+08  4.105521e+08              0.456522          NaN  
std         0.0  1.211677e+09  1.211677e+09              1.592989          NaN  
min         0.0  2.118613e+03  2.118613e+03              0.000000          NaN  
25%         0.0  7.636318e+04  7.636318e+04              0.000000          NaN  
50%         0.0  7.814786e+05  7.814786e+05              0.000000          NaN  
75%         0.0  3.979433e+07  3.979433e+07              0.000000          NaN  
max         0.0  7.654650e+09  7.654650e+09              6.000000          NaN  

[8 rows x 22 columns]

* General statistics about dataset. It also include outlier constants.

In [44]:
df.isnull().sum()

TypeCode                      0
FreqCode                      0
RefPeriodId                   0
RefYear                       0
RefMonth                      0
Period                        0
ReporterCode                  0
ReporterISO                   0
ReporterDesc                  0
FlowCode                      0
FlowDesc                      0
PartnerCode                   0
PartnerISO                    0
PartnerDesc                   0
Partner2Code                  0
Partner2ISO                   0
Partner2Desc                  0
ClassificationCode            0
ClassificationSearchCode      0
IsOriginalClassification      0
CmdCode                       0
CmdDesc                       0
AggrLevel                     0
IsLeaf                        0
CustomsCode                   0
CustomsDesc                   0
MosCode                       0
MotCode                       0
MotDesc                       0
QtyUnitCode                   0
QtyUnitAbbr                  21
Qty     

* One of the most important function that gives NaN values in the dataset. As we can see that **Unnamed: 47** is unnecessery column and the other one can be **CifValue**.